js_modules/dagster-ui/packages/ui-components/src/components/SplitPanelContainer.tsx

```diff
@@ -13,220 +11,185 @@ interface SplitPanelContainerProps {
   first: React.ReactNode;
   firstInitialPercent: number;
   firstMinSize?: number;
+  secondMinSize?: number;
   second: React.ReactNode | null; // Note: pass null to hide / animate away the second panel
 }
 
-interface SplitPanelContainerState {
-  size: number;
-  key: string;
-  resizing: boolean;
-}
-
-export class SplitPanelContainer extends React.Component<
-  SplitPanelContainerProps,
-  SplitPanelContainerState
-> {
-  constructor(props: SplitPanelContainerProps) {
-    super(props);
-
-    const key = `dagster.panel-width.${this.props.identifier}`;
-    const value = window.localStorage.getItem(key);
-    let size = Number(value);
-    if (value === null || isNaN(size)) {
-      size = this.props.firstInitialPercent;
-    }
-
-    this.state = {size, key, resizing: false};
-  }
+export type SplitPanelContainerHandle = {
+  getSize: () => number;
+  changeSize: (value: number) => void;
+};
 
-  onChangeSize = (size: number) => {
-    this.setState({size});
-    window.localStorage.setItem(this.state.key, `${size}`);
-  };
+export const SplitPanelContainer = forwardRef<SplitPanelContainerHandle, SplitPanelContainerProps>(
+  (props, ref) => {
+    const {
+      axis = 'horizontal',
+      identifier,
+      first,
+      firstInitialPercent,
+      firstMinSize,
+      secondMinSize = 0,
+      second,
+    } = props;
+
+    const [_, setTrigger] = useState(0);
+    const [resizing, setResizing] = useState(false);
+    const key = `dagster.panel-width.${identifier}`;
+
+    const getSize = useCallback(() => {
+      if (!second) {
+        return 100;
+      }
+      const storedSize = window.localStorage.getItem(key);
+      const parsed = storedSize === null ? null : parseFloat(storedSize);
+      return parsed === null || isNaN(parsed) ? firstInitialPercent : parsed;
+    }, [firstInitialPercent, key, second]);
+
+    const onChangeSize = useCallback(
+      (newValue: number) => {
+        window.localStorage.setItem(key, `${newValue}`);
+        setTrigger((current) => (current ? 0 : 1));
```

Comment:
We're already tracking the `size` value in localStorage, so we don't need to duplicate that in local state.